Source/CarthageKit/BuildArguments.swift

```diff
@@ -84,6 +84,11 @@ public struct BuildArguments {
 			args += [ "BITCODE_GENERATION_MODE=\(bitcodeGenerationMode.rawValue)" ]
 		}
 
+		if let carthagePath = NSBundle.mainBundle().executablePath
+			where NSBundle.mainBundle().executableURL?.lastPathComponent?.lowercaseString.rangeOfString("carthage") != nil {
```

Comment:
That's still unnecessary work. Why construct a new lowercase string when it's trivial to just do a lowercase compare? I still disagree about doing `rangeOfString`, because people don't typically rename executables, but if you really want to, then you can use `rangeOfString("carthage", options: [.CaseInsensitiveSearch]) != nil`.